### PR TITLE
*: Save Xenon's metadata to config map.

### LINKED ISSUE #406: [feature] Save Xenon's metadata to persistent storage.
**Is your feature request related to a problem? Please describe.**

The physical machine restart will need to re-load Xenon metadata by Operator, because Xenon's metadata directory(`/var/lib/xenon/`) is not persistent storage.
```
cat /var/lib/xenon/peers.json
{"idlepeers":null,"peers":["sample-mysql-1.sample-mysql.default:8801","sample-mysql-0.sample-mysql.default:8801","sample-mysql-2.sample-mysql.default:8801"]}
```
**Describe the solution you'd like**

We can directly store Xenon's metadata persistence in disk or configmap.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by runkecheng**
### What type of PR is this?

/enhancement

### Which issue(s) this PR fixes?

Fixes #406 #439

### What this PR does?

Summary:

subpath -> configmap
subpath may not work in some conditions. see #439 
1. Rename configMapSyncer to mysqlCMSyncer.
2. Add a Xenon Configmap which is used to save xenon data.
3. Build peers.json according to the replicas.
4. Make peers.json in init-sidecar.
5. Xenon boot using existing peers.json.

### CODE REVIEW

**Review by andyli029 — COMMENTED** [review]
Add #439 to the commit log

File: controllers/mysqlcluster_controller.go [hunk 1]
```diff
@@ -109,7 +109,8 @@ func (r *MysqlClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
-	cmRev := configMapSyncer.Object().(*corev1.ConfigMap).ResourceVersion
+	// Need fix: modify mysql cm will trigger rolling update but it will not be applied.
```
Comment (andyli029) [inline controllers/mysqlcluster_controller.go:112]: Need fix
->
TODO:

File: mysqlcluster/container/xenon.go [hunk 2]
```diff
@@ -118,6 +118,10 @@ func (c *xenon) getVolumeMounts() []corev1.VolumeMount {
 			Name:      utils.XenonVolumeName,
 			MountPath: utils.XenonVolumeMountPath,
 		},
+		{
+			Name:      utils.XenonMetaVolumeName,
```
Comment (andyli029) [inline mysqlcluster/container/xenon.go:122]: Name:      utils.XenonMetaVolumeName,
			MountPath: utils.XenonConfVolumeMountPath,
->
			Name:      utils.XenonMetaVolumeName,
			MountPath: utils.XenonMetaVolumeMountPath,
better?

File: mysqlcluster/container/xenon_test.go [hunk 3]
```diff
@@ -138,6 +138,10 @@ func TestGetXenonVolumeMounts(t *testing.T) {
 			Name:      "xenon",
 			MountPath: "/etc/xenon",
 		},
+		{
```
Comment (andyli029) [inline mysqlcluster/container/xenon_test.go:141]:
```
xenon-meta
different between
xenon
```

Comment (runkecheng) [reply mysqlcluster/container/xenon_test.go:141]: /etc/xenon save xenon config, var/lib/xenon save xenon metadata.

**Review by andyli029 — COMMENTED** [review]
LGTM.

**Pull request merged**